zcash_proofs/src/sapling/prover.rs

```diff
@@ -154,15 +154,7 @@ impl SaplingProvingContext {
         }
 
         // Verify the proof
-        match verify_proof(verifying_key, &proof, &public_input[..]) {
-            // No error, and proof verification successful
-            Ok(true) => {}
-
-            // Any other case
-            _ => {
-                return Err(());
-            }
-        }
+        verify_proof(verifying_key, &proof, &public_input[..]).map_err(|_| ())?;
```

Comment:
It'd be nice if the whole spend_proof function returned a meaningful error type rather than just throwing away all the information it could be providing to the caller. Just a note for future work.